src/test/java/org/springframework/data/couchbase/repository/CouchbaseRepositoryKeyValueIntegrationTests.java

```diff
@@ -54,7 +51,16 @@
 
 	@Autowired UserRepository userRepository;
 	@Autowired LibraryRepository libraryRepository;
-	@Autowired UserSubmissionRepository userSubmissionRepository;
+	@Autowired SubscriptionTokenRepository subscriptionTokenRepository;
+
+	@Test
+	void subscriptionToken(){
+		SubscriptionToken st = new SubscriptionToken("id", 0, "type", "Dave Smith",
+				"app123", "dev123", 0);
+		st = subscriptionTokenRepository.save(st);
+		st = subscriptionTokenRepository.findById(st.getId()).get();
```

Comment:
can you assert that the cas stored in the token is the same as when read "raw" from the SDK to make sure it is populated properly?

Reply:
Ok, will add that check